website/src/views/DashboardRouter/DashboardRouter.jsx

```diff
@@ -70,24 +73,87 @@ class DashboardRouter extends PureComponent {
     });
   };
 
+  viewEvent = (eventData) => {
+    if (navigator.onLine) {
+      message.info('Getting event...');
+      const { userData } = this.state;
+      const { eventUid, classUid, university } = eventData;
+      const { uid } = firebase.auth().currentUser;
+      const reqData = { eventUid, classUid, uid };
+      fetch(
+        'https://us-central1-rails-students.cloudfunctions.net/geteventdetails',
+        {
+          method: 'POST',
+          headers: {
+            Accept: 'application/json',
+            'Content-Type': 'application/json',
+          },
+          body: JSON.stringify(reqData),
+        },
+      )
+        .then(res => res.json())
+        .then((result) => {
+          message.info(result.message);
+          console.log(result);
+          if (result.message === 'Found event') {
+            result.eventData.university = university;
+            result.eventData.classUid = classUid;
+            result.eventData.eventUid = eventUid;
+            if (userData.type === 'teacher' && (result.eventData.instructorUid === uid)) {
+              this.setState({
+                visible: true,
+                eventData: result.eventData,
```

Comment:
You should cache the eventdata to localstorage for retrieval later

Reply:
not safe because if teacher, sensitive data is fetched. could be exploited. best not to cache sensitive data.

Reply:
Then that kind of defeats the point of it being a PWA if you cant access any of the data while offline.

Reply:
Security over convenience?

Reply:
And its saved locally to localstorage websites cant access each others localstorage. https://stackoverflow.com/questions/12437836/html5-web-storage-can-different-websites-overwrite-each-other-s-data-on-a-user 
Only way to exploit is if an attacker has physical access to the victims browser to view  content.

Reply:
@khelif96 let's stick with the features we have. last minute rush leads to lower quality features.